### PR TITLE
feat: privacy and push notifications settings

### PR DESCRIPTION
**Pull request by ice-brontes**
## Description
Add settings pages for privacy and push notifications sections.

## Additional Notes
1. Added app name to the strings inside `SettingsRedirectSheet` where it was used.
2. Added settings pages for privacy and push notifications.
3. Updated [icon_block_checkbox_on.svg](https://github.com/ice-blockchain/ion-app/pull/317/files#diff-a8ceef03a9534ecfebef0de028e2d8905e1c007640dd10657b6c7d51ee0e5e5c) so it can be colorized properly.
4. Configured `SettingsRedirectSheet` and `PermissionRequestSheet` so they can work with notifications permission.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation
- [ ] Chore

## Video demonstration
https://github.com/user-attachments/assets/d1afb946-8093-46c7-a712-aea6dc7394f1

## Figma Link (if applicable)
[Design](https://www.figma.com/design/3TnZngoFeklOVC8TexlR1D/%F0%9F%A7%8A-ICE-Cloud?node-id=44447-189504&m=dev)

### CODE REVIEW

**Review by ice-kreios — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

File: lib/app/features/settings/components/selectable_options_group.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        SizedBox(height: 16.0.s),
+        SeparatedColumn(
+          mainAxisSize: MainAxisSize.min,
+          separator: Padding(
+            padding: EdgeInsets.symmetric(vertical: 12.0.s),
+            child: const HorizontalSeparator(),
+          ),
+          children: options
+              .map(
+                (e) => _OptionItem(
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/components/selectable_options_group.dart]: Done.

File: lib/app/features/settings/components/selectable_options_group.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  final String title;
+  final List<T> options;
+  final List<T> selected;
+  final void Function(T) onSelected;
+  final bool addSeparatorAfterLastItem;
+  final bool enabled;
+
+  @override
+  Widget build(BuildContext context) {
+    final separatorPadding = EdgeInsets.symmetric(vertical: 12.0.s);
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/components/selectable_options_group.dart]: Removed.

File: lib/app/features/settings/views/privacy_settings_modal.dart [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            actions: [
+              NavigationCloseButton(
+                onPressed: Navigator.of(context, rootNavigator: true).pop,
+              ),
+            ],
+          ),
+          ScreenSideOffset.small(
+            child: SeparatedColumn(
+              mainAxisSize: MainAxisSize.min,
+              separator: SizedBox(height: 16.0.s),
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/views/privacy_settings_modal.dart]: Removed this flag and used the `SelectableOptionsGroup` separator.

File: lib/app/features/components/avatar_picker/avatar_picker.dart [hunk 4]
```diff
@@ -58,7 +59,11 @@ class AvatarPicker extends ConsumerWidget {
                     );
               }
             },
-            requestDialog: PermissionRequestSheet.fromType(context, Permission.camera),
+            requestDialog: PermissionRequestSheet.fromType(
+              context,
+              permissionType: Permission.camera,
+              appName: ref.watch(appInfoProvider).valueOrNull?.appName ?? '',
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/components/avatar_picker/avatar_picker.dart]: So, I just passed `Permission` to the `PermissionRequestSheet` and in the build method created `InfoCard` widget according to the permission type. I tried all the suggested solutions and this one required the least amount of code and allowed to use `const` when creating the widget.

File: lib/app/features/settings/components/selectable_options_group.dart [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(title, style: context.theme.appTextThemes.body),
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/components/selectable_options_group.dart]: Done

File: lib/app/features/settings/views/privacy_settings_modal.dart [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import 'package:ion/app/features/settings/model/privacy_options.dart';
+import 'package:ion/app/router/components/navigation_app_bar/navigation_app_bar.dart';
+import 'package:ion/app/router/components/navigation_app_bar/navigation_close_button.dart';
+import 'package:ion/app/router/components/sheet_content/sheet_content.dart';
+
+class PrivacySettingsModal extends HookConsumerWidget {
+  const PrivacySettingsModal({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/views/privacy_settings_modal.dart]: Removed all unused `ref`s.

File: lib/app/features/settings/views/push_notifications_settings.dart [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    required this.hasPermissionOnDeviceLevel,
+    required this.onChangeAppLevelPermission,
+  });
+
+  final VoidCallback onChangeAppLevelPermission;
+  final bool hasPermissionOnDeviceLevel;
+  final bool hasPermission;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/views/push_notifications_settings.dart:131]: It used to watch on notifications' permission. 

`final hasPermissionOnDeviceLevel = ref.watch(hasPermissionProvider(Permission.notifications));`

File: lib/app/features/feed/views/pages/feed_main_modal/feed_main_modal_page.dart [hunk 8]
```diff
@@ -13,13 +14,13 @@ import 'package:ion/app/router/components/navigation_app_bar/navigation_app_bar.
...
 
-class FeedMainModalPage extends StatelessWidget {
+class FeedMainModalPage extends ConsumerWidget {
   const FeedMainModalPage({super.key});
 
   static const List<FeedType> feedTypeValues = FeedType.values;
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/feed/views/pages/feed_main_modal/feed_main_modal_page.dart]: Removed.

File: lib/app/features/settings/views/push_notifications_settings.dart (hunk 7 above)
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/settings/views/push_notifications_settings.dart]: Oh... I see, removed.

**Review by ice-kreios — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**